node/coinstacks/ethereum/api/src/controller.ts

```diff
@@ -147,29 +147,58 @@ export class Ethereum extends Controller implements BaseAPI, EthereumAPI {
   @Get('account/{pubkey}/txs')
   async getTxHistory(
     @Path() pubkey: string,
-    @Query() page?: number,
-    @Query() pageSize = 25,
-    @Query() contract?: string
-  ): Promise<TxHistory> {
+    @Query() cursor?: string,
+    @Query() pageSize = 25
+  ): Promise<EthereumTxHistory> {
     try {
-      const data = await blockbook.getAddress(pubkey, page, pageSize, undefined, undefined, 'txs', contract)
+      let page = 1
+      if (cursor) {
+        const cursorDecoded = Buffer.from(cursor, 'base64').toString('binary')
+        if (isNaN(Number(cursorDecoded))) {
+          throw new ApiError('Bad Request', 400, 'Invalid cursor, please use the cursor returned in previous page')
+        }
+        page = Number(cursorDecoded)
+      }
+
+      const data = await blockbook.getAddress(pubkey, page, pageSize, undefined, undefined, 'txs')
+
+      const nextPage = page + 1
+
+      const nextCursor =
+        (data.transactions?.length ?? 0) == pageSize
+          ? { cursor: Buffer.from(nextPage.toString(), 'binary').toString('base64') }
+          : undefined
 
       return {
-        page: data.page ?? 1,
-        totalPages: data.totalPages ?? 1,
-        txs: data.txs,
-        transactions:
-          data.transactions?.map<Tx>((tx) => ({
+        pubkey: pubkey,
+        ...nextCursor,
+        txs:
+          data.transactions?.map<EthereumTx>((tx) => ({
             txid: tx.txid,
-            status: tx.confirmations > 0 ? 'confirmed' : 'pending',
-            from: tx.vin[0].addresses?.[0] ?? 'coinbase',
-            to: tx.vout[0].addresses?.[0],
             blockHash: tx.blockHash,
             blockHeight: tx.blockHeight,
-            confirmations: tx.confirmations,
             timestamp: tx.blockTime,
+            status: tx.ethereumSpecific?.status ?? tx.confirmations > 0 ? 1 : -1, // assuming confirmed=1 pending=-1 based on previous implementation and TransactionParser.getStatus()
+            from: tx.vin[0].addresses?.[0] ?? 'coinbase',
```

Comment:
Lets go with empty string for all of these default values

Reply:
Here used empty string for all the default values, also for the from field which was 'coinbase' in the previous implementation.